content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md

```diff
@@ -5,6 +5,56 @@ weight=99
 
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+Labels are key-value pairs, stored as strings, that are attached to an image (i.e., arbitrary metadata). Labels are used to add helpful descriptions or attributes to an application image, which are meaningful to users.
 
-If you are familiar with this content and would like to make a contribution, please feel free to open a PR :)
+Labels are usually added at the time an image is created and can also be modified at a later time. Images can have multiple labels; however, each key must be unique.
+
+## Key Points
+
+A `LABEL`
+
+* MUST specify a unique key for a given image
+* MUST specify a value to be set in the image label  
+* MUST be added as an image label on the created image metadata
+* If one key is given multiple values, the last value overwrites  
+
+## Use Cases
+
+Since adding labels to application images is optional and not needed to run containers, this property is often overlooked. The following are few reasons to why labels should be widely adopted  
+
+* Documenting versioning
+* Record licensing information
+* Including information about a project maintainer
+* Including a description of the image and  additional metadata related to the image
+* Labels can also be used to organize images  
+
+## Implementation Steps
+
+* General syntax of `LABEL` instruction is as follows:
+
+    `LABEL <key-string>=<value-string>`
+
+    `LABEL version="1.0"`
```

Comment:
Taking the perspective of a buildpack author, they would need to write 

```toml
[[labels]]
key-string = "value-string"
otherkey = "othervalue"
```

to `launch.toml` in their `<layers>/<layer>` directory.  If we use the running example from our Buildpack Author's Guide, this means writing to `"${CNB_LAYERS_DIR}"/node-js/launch.toml`.  However, it's worth noting that `launch.toml` also contains metadata about process and slices.

Reply:
not sure I get this? apologies